src/components/Header.tsx

```diff
@@ -8,9 +8,14 @@ import HeaderProfile from './HeaderProfile'
 import { useIsLoggedIn } from '../utils/hooks'
 import { useState } from 'react'
 import Sidebar from './Sidebar'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 
 const Header = (): JSX.Element => {
   const location = useLocation()
+
+  const paths = location.pathname.split('/')
+  const id = paths[2] === 'edit' ? 'NaN' : parseInt(paths[paths.length - 1], 10)
```

Comment:
This feels very hacky, can't we retrieve the ID from some context manager instead?